protocol.go

```diff
@@ -394,3 +397,24 @@ func canonicalizeContentTypeSlow(contentType string) string {
 	}
 	return mime.FormatMediaType(base, params)
 }
+
+// defaultInitializer is the default initializer for dynamic messages. It
+// initializes the message to the type specified in the Spec.
+func defaultInitializer(spec Spec, msg any) error {
```

Comment:
This is the current default initializer that support `dynamicpb.Message`. Do we want this? Or only explicitly providing an initializer? Having this as the default will always interface cast the type.

Reply:
We decided to provide this as an option as an alternative to adding transparent internal support for `dynamicpb`. So I don't think we need to do both. The default can be nil, which simply means no initialization needed.